backend/lib/spark-api-deal-submitter.js

```
@@ -0,0 +1,127 @@
+/** @import {PgPool, Queryable} from '@filecoin-station/deal-observer-db' */
+import Cursor from 'pg-cursor'
+
+/**
+ * Finds deals that haven't been submitted to the Spark API yet and submits them.
+ *
+ * @param {PgPool} pgPool
+ * @param {number} batchSize
+ * @param {(eligibleDeals: Array) => Promise<void>} submitDeals
+ */
+export const findAndSubmitUnsubmittedDeals = async (pgPool, batchSize, submitDeals) => {
+  console.debug(`Finding and submitting deals using batchSize: ${batchSize}`)
+  for await (const unsubmittedDeals of findUnsubmittedDeals(pgPool, batchSize)) {
+    console.debug(`Found ${unsubmittedDeals.length} unsubmitted deals`)
+    try {
+      const formattedDeals = formatUnsubmittedDealsForSparkApi(unsubmittedDeals)
+      await submitDeals(formattedDeals)
+      console.debug(`Successfully submitted ${formattedDeals.length} deals`)
+      await markDealsAsSubmitted(pgPool, unsubmittedDeals)
+    } catch (e) {
+      console.error('Failed to submit deals:', e)
```

Should we report the error to Sentry?